packages/docs/community/release-process/blockchain-client.md

```diff
@@ -0,0 +1,179 @@
+
+```bash
+gpg --verify celo-blockchain-vX.Y.Z-stable.tar.gz.asc celo-blockchain-vX.Y.Z-stable.tar.gz
+```
+
+### Docker images
+Docker images are built automatically with [Google Cloud Build](https://cloud.google.com/cloud-build) upon pushes to `master` and all release branches. Automated builds will be tagged in [Google Container Registry](https://cloud.google.com/container-registry) with the corresponding commit hash.
+
+A signature should be produced over the image automatically built at the corresponding commit hash and included with the Github release.
+
+Release image signatures can be verified with the following command:
+
+```bash
+gpg --verify celo-blockchain-vX.Y.Z.docker.asc <(docker save us.gcr.io/celo-testnet/celo-node:vX.Y.Z)
+```
+
+## Testing
+
+All builds of `celo-blockchain` are automatically tested for performance and backwards compatibility in CI. Any regressions in these tests should be considered a blocker for a release.
+
+Minor and major releases are expected to go through additional rounds of manual testing as needed to verify behavior under stress conditions, such as a network with faulty nodes, and poor network connectivity.
+
+## Promotion process
+
+### Source control
+Patch releases should be constructed by cherry-picking all included commits from `master` to the `release/x.y` branch. The first commit of this process should change the version number encoded in the source from `x.y.z-stable` to `x.y.z+1-unstable` and the final commit should change the version number to `x.y.z+1-stable`.
+
+Major and minor releases should be constructed by pushing a commit to the `master` branch to change the encoded version number from `x.y.z-unstable` to `x.y.z-stable`. A `release/x.y` branch should be created from this commit.  The next commit must change the version number from `x.y.z-stable` to `x.y+1.0-unstable`, or `x+1.0.0-unstable` if the next planned release is a major release.
+
+Only one commit should ever have a “stable” tag at any given version number. When that commit is created, a tag should be added along with release notes. Once the tag is published it should not be reused for any further release or changes.
+
+### Distribution
+Distribution of an image should occur along the following schedule:
+
+<table>
+  <tr>
+    <td>Date</td>
+    <td>Action</td>
+  </tr>
+  <tr>
+    <td>T</td>
+    <td>
+      <ol>
+        <li>Publish the Git tag and signed release artifacts.</li>
+        <li>Communicate T+1w Baklava upgrade date.</li>
+        <li>Tag released Docker image with <code>baklava</code>.</li>
+      </ol>
+    </td>
+  </tr>
+  <tr>
+    <td>T+1w</td>
+    <td>
+      <ol>
+        <li>Confirm Baklava users have upgraded without issues</li>
+        <li>If release introduces a hard fork</li>
+        <ol>
+          <li>Ensure at least a quorum of the validator set has upgraded</li>
+          <li>Submit governance proposal to increment minimum client version</li>
```

Comment:
I think we might need to add some additional clarification here. I'm still a bit murky on what it means to introduce a hard fork. I might have some faulty assumptions so let me do a little brain dump.

As far as I understand it there's a `minimum client version` stored on chain that, I imagine, acts as a circuit breaker and makes the node go "I can't play with others so I'm shutting down". But in order for the network to fork you'd also need another off-chain `minimum client version` that tells the client what peers she can play with, thus an update that increments this off-chain constraint will cause the nodes that update to play among themselves. 
But at this point we have two chains so the governance proposal seams useless because it essentially happens on the chain that already is constrained by the off-chain version. And the old chain will not necessarily get the proposal and continue to function independently, albeit without any use, but without the nodes affected by the on-chain `minimum client version`. 

So I feel like I'm miss-understanding something and have some faulty assumptions, either way I think it would be valuable to expand on these scenarios better here. Explain how a release that introduces a hard fork happens vs one that doesn't.

I guess this also started from me trying to understand why incrementing happened _only_ in hard fork releases.

Reply:
Was a miss-understanding on my part which was resolved in call with @nategraf.
I'm still unsure wether we should include more information here about this or leave it as is.

Reply:
I am personally in favor of leaving it at the current level of specificity because I think we will understand better how to handle handforks once we are used to doing non-fork upgrades. Additionally, there may be multiple processes that would be used at any time, and this PR is designed to get use to an initial implementation, and does not have to be complete or perfect.